src/cargo/ops/cargo_new.rs

```diff
@@ -412,73 +489,46 @@ fn mk(config: &Config, opts: &MkOptions) -> CargoResult<()> {
         (None, None, name, None) => name,
     };
 
-    let mut cargotoml_path_specifier = String::new();
-
-    // Calculare what [lib] and [[bin]]s do we need to append to Cargo.toml
-
-    for i in &opts.source_files {
-        if i.bin {
-            if i.relative_path != "src/main.rs" {
-                cargotoml_path_specifier.push_str(&format!(r#"
-[[bin]]
-name = "{}"
-path = {}
-"#, i.target_name, toml::Value::String(i.relative_path.clone())));
-            }
-        } else {
-            if i.relative_path != "src/lib.rs" {
-                cargotoml_path_specifier.push_str(&format!(r#"
-[lib]
-name = "{}"
-path = {}
-"#, i.target_name, toml::Value::String(i.relative_path.clone())));
-            }
+    // construct the mapping used to populate the template
+    // if in the future we want to make more varaibles available in
+    // the templates, this would be the place to do it.
+    let mut handlebars = Handlebars::new();
+    // We don't want html escaping unless users explicitly ask for it...
+    handlebars.register_escape_fn(no_escape);
+    handlebars.register_helper("toml-escape", Box::new(template::toml_escape_helper));
+    handlebars.register_helper("html-escape", Box::new(template::html_escape_helper));
+
+    let mut data = BTreeMap::new();
+    data.insert("name".to_owned(), name.to_owned());
+    data.insert("author".to_owned(), author);
+
+    let template_set = try!(get_input_template(config, opts));
+    for template in template_set.template_files.iter() {
+        let template_str = try!(template.template());
+        //let dest_file_name = try!(abs_to_rel_path(entry_path, template_path));
+        //let mut dest_path = PathBuf::from(path).join(dest_file_name);
+        let dest_path = PathBuf::from(path).join(template.path());
+
+        // Skip files that already exist.
+        if fs::metadata(&dest_path).is_ok() {
+            continue;
         }
-    }
-
-    // Create Cargo.toml file with necessary [lib] and [[bin]] sections, if needed
-
-    try!(paths::write(&path.join("Cargo.toml"), format!(
-r#"[package]
-name = "{}"
-version = "0.1.0"
-authors = [{}]
-
-[dependencies]
-{}"#, name, toml::Value::String(author), cargotoml_path_specifier).as_bytes()));
 
+        create_deep_subdirs(&dest_path);
 
-    // Create all specified source files
-    // (with respective parent directories)
-    // if they are don't exist
-
-    for i in &opts.source_files {
-        let path_of_source_file = path.join(i.relative_path.clone());
-
-        if let Some(src_dir) = path_of_source_file.parent() {
-            try!(fs::create_dir_all(src_dir));
-        }
+        // create the new file & render the template to it
+        let mut dest_file = try!(OpenOptions::new().write(true)
+                                 .create(true)
```

Comment:
I think this can just be `File::create` rather than going through `OpenOptions`, right?